executor/admin.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/ranger"
+	"github.com/pingcap/tipb/go-tipb"
+	"golang.org/x/net/context"
+)
+
+var _ Executor = new(CheckIndexRangeExec)
+
+// CheckIndexRangeExec output the index values which has handle between begin and end.
+type CheckIndexRangeExec struct {
+	baseExecutor
+
+	table    table.Table
+	index    table.Index
+	is       infoschema.InfoSchema
+	startKey []types.Datum
+
+	handleRanges []ast.HandleRange
+	srcChunk     *chunk.Chunk
+
+	result   distsql.SelectResult
+	partial  distsql.PartialResult
+	typesMap map[int64]*types.FieldType
+	cols     []*model.ColumnInfo
+}
+
+// NextChunk implements the Executor Next interface.
+func (e *CheckIndexRangeExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	handleIdx := e.schema.Len() - 1
+	for {
+		err := e.result.NextChunk(ctx, e.srcChunk)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if e.srcChunk.NumRows() == 0 {
+			return nil
+		}
+		for i := 0; i < e.srcChunk.NumRows(); i++ {
+			row := e.srcChunk.GetRow(i)
+			handle := row.GetInt64(handleIdx)
+			for _, hr := range e.handleRanges {
+				if handle >= hr.Begin && handle < hr.End {
+					chk.AppendRow(row)
+				}
+			}
+		}
+		if chk.NumRows() >= e.maxChunkSize {
+			return nil
+		}
+	}
+}
+
+// Next implements the Executor Next interface.
+func (e *CheckIndexRangeExec) Next(ctx context.Context) (Row, error) {
+	return nil, nil
+}
+
+// Open implements the Executor Open interface.
+func (e *CheckIndexRangeExec) Open(ctx context.Context) error {
+	e.typesMap = make(map[int64]*types.FieldType)
+	tCols := e.table.Meta().Cols()
+	for _, ic := range e.index.Meta().Columns {
+		col := tCols[ic.Offset]
+		e.typesMap[col.ID] = &col.FieldType
+		e.cols = append(e.cols, col)
+	}
+	e.srcChunk = e.newChunk()
+	e.retFieldTypes = make([]*types.FieldType, e.schema.Len())
+	for i, v := range e.schema.Columns {
+		e.retFieldTypes[i] = v.RetType
+	}
+	dagPB, err := e.buildDagPB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	sc := e.ctx.GetSessionVars().StmtCtx
+	var builder distsql.RequestBuilder
+	kvReq, err := builder.SetIndexRanges(sc, e.table.Meta().ID, e.index.Meta().ID, ranger.FullNewRange()).
+		SetDAGRequest(dagPB).
+		SetKeepOrder(true).
+		SetFromSessionVars(e.ctx.GetSessionVars()).
+		Build()
+
+	e.result, err = distsql.Select(ctx, e.ctx, kvReq, e.retFieldTypes)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	e.result.Fetch(ctx)
+	return nil
+}
+
+func (e *CheckIndexRangeExec) buildDagPB() (*tipb.DAGRequest, error) {
+	dagReq := &tipb.DAGRequest{}
+	dagReq.StartTs = e.ctx.Txn().StartTS()
+	dagReq.TimeZoneOffset = timeZoneOffset(e.ctx)
+	sc := e.ctx.GetSessionVars().StmtCtx
+	dagReq.Flags = statementContextToFlags(sc)
```

Comment:
Could we use `constructDAGReq` directly?
And can we put this function of `buildDagPB` to `builder.go` as other executors?
s/buildDagPB/buildDAGPB or buildDAGReq ?

Reply:
No, we don't have a PhysicalPlan here.
It's not a general way to build a DAG request.

Reply:
We can make some util to easily build DAG requests for both executor and DDL later.